### DISCUSSION TITLE
New Locking and pgbouncer

### DISCUSSION THREAD

**Discussion by thegcat** [pretix/pretix#3620] [Q&A] [answered]
Is there any special care needed when using Pretix 2023.8+ with the new locking mechanism https://github.com/pretix/pretix/pull/2408 when using pgbouncer?

We already ran into weirdness with pgbouncer and advisory locks (e.g. Ruby on Rails now uses advisory locks for migrations to ensure only one instance/worker runs migrations at any given time, but our pgbouncer seemingly had distributed its one incoming connection to several connections to the DB, which caused the lock release to fail as it didn't come from the same connection as seen by the DB), so any experience you have for this already would be appreciated, thank you.

**Comment by raphaelm** [comment, marked as answer]
Advisory locks in PostgreSQL exist both on transaction level and on session level. We only use transaction level, therefore I believe there should be no issue in the transaction pooling mode of pgbouncer. We have not tested it, though, as we are not running pgbouncer so far (PostgreSQL can happily handle around 1k "real" connections before things get slow).

**Reply by raphaelm** [reply]
mistyped my message to mean the opposite, i have corrected it now above

**Reply by raphaelm** [reply]
mistyped my message to mean the opposite, i have corrected it now above

**Comment by thegcat** [comment]
Thank you. We'll run it as is and let you know should we run into any weird things nonetheless.
